src/event/metric.rs

```diff
@@ -215,6 +217,121 @@ impl Metric {
+    /// ```
+    fn fmt(&self, fmt: &mut Formatter<'_>) -> Result<(), fmt::Error> {
+        if let Some(timestamp) = &self.timestamp {
+            write!(fmt, "{:?} ", timestamp)?;
+        }
+        write!(fmt, "{}", quote_word(&self.name))?;
+        if let Some(tags) = &self.tags {
+            write!(fmt, "{{")?;
+            write_list(fmt, ",", tags.iter(), |fmt, (tag, value)| {
+                write!(fmt, "{}={:?}", quote_word(tag), value)
+            })?;
+            write!(fmt, "}}")?;
+        }
+        match self.kind {
+            MetricKind::Absolute => write!(fmt, " = "),
+            MetricKind::Incremental => write!(fmt, " + "),
+        }?;
+        match &self.value {
+            MetricValue::Counter { value } => write!(fmt, "{}", value),
+            MetricValue::Gauge { value } => write!(fmt, "{}", value),
+            MetricValue::Set { values } => write_list(fmt, " ", values.iter(), |fmt, value| {
+                write!(fmt, "{}", quote_word(value))
+            }),
+            MetricValue::Distribution {
+                values,
+                sample_rates,
+                statistic,
+            } => {
+                write!(fmt, "{:?} ", statistic)?;
+                write_list(
+                    fmt,
+                    " ",
+                    values.iter().zip(sample_rates.iter()),
+                    |fmt, (value, rate)| write!(fmt, "{}@{}", rate, value),
+                )
+            }
+            MetricValue::AggregatedHistogram {
+                buckets,
+                counts,
+                count,
+                sum,
+            } => {
+                write!(fmt, "count={} sum={} ", count, sum)?;
+                write_list(
+                    fmt,
+                    " ",
+                    buckets.iter().zip(counts.iter()),
+                    |fmt, (bucket, count)| write!(fmt, "{}@{}", bucket, count),
+                )
+            }
+            MetricValue::AggregatedSummary {
+                quantiles,
+                values,
+                count,
+                sum,
+            } => {
+                write!(fmt, "count={} sum={} ", count, sum)?;
+                write_list(
+                    fmt,
+                    " ",
+                    quantiles.iter().zip(values.iter()),
+                    |fmt, (quantile, value)| write!(fmt, "{}@{}", quantile, value),
+                )
+            }
+        }
+    }
+}
+
+fn write_list<I, T, W>(
+    fmt: &mut Formatter<'_>,
+    sep: &str,
+    items: I,
+    writer: W,
+) -> Result<(), fmt::Error>
+where
+    I: Iterator<Item = T>,
+    W: Fn(&mut Formatter<'_>, T) -> Result<(), fmt::Error>,
+{
+    let mut this_sep = "";
+    for item in items {
+        write!(fmt, "{}", this_sep)?;
+        writer(fmt, item)?;
+        this_sep = sep;
+    }
+    Ok(())
+}
+
+fn quote_word(s: &str) -> Cow<'_, str> {
+    const ALPHANUM: &str = "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789_";
```

Comment:
Why not `char::is_alphanumeric`?

Reply:
Mostly because I was ignorant of that method. However, that method also matches non-ASCII alphabetic characters, which might be undesirable. `char::is_ascii_alphanumeric` would be better, though I'll also have to check for underscore.

Reply:
Then `is_ascii_alphanumeric` and a check for underscore seem reasonable. It's faster and clearer.